app/parsers/concerns/hyku_addons/collection_behavior.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+module HykuAddons
+  module CollectionBehavior
+    def collections
+      records.map do |r|
+        collection_ids(r)&.each_with_index&.map do |id, i|
+          if r[collection_title_key].present?
+            { id: collection_ids(r)[i], title: collection_titles(r)[i] }
+          else
+            { id: id, title: "New collection #{i + 1}" }
+          end
+        end
+      end.flatten.compact.uniq
+    end
+
+    # Override to pass Bulkrax.system_identifier_field as singular instead of array
+    def create_collections
+      admin_sets.each_with_index do |admin_set_name, index|
+        call_collection_job(admin_set_entry_class, admin_set_name, admin_set_metadata(admin_set_name))
+
+        increment_counters(index, true)
+      end
+
+      collections.each_with_index do |collection, index|
+        call_collection_job(collection_entry_class, collection[:id], collection_metadata(collection))
+
+        increment_counters(index, true)
+      end
+    end
+
+    def collections_total
+      collections.size + admin_sets.size
+    end
+
+    private
+
+      def call_collection_job(item_entry_class, item_id, item_metadata)
+        return if item_id.empty?
+
+        new_entry = find_or_create_entry(item_entry_class, item_id, 'Bulkrax::Importer', item_metadata)
+
+        begin
+          Bulkrax::ImportWorkCollectionJob.perform_now(new_entry.id, current_run.id)
+        rescue StandardError => e
+          new_entry.status_info(e)
+        end
+      end
+
+      def admin_set_metadata(admin_set)
+        {
+          title: [admin_set],
+          # Allow Hyku to generate a UUID for the admin set
+          # Bulkrax.system_identifier_field => nil,
+          visibility: 'open',
+          collection_type_gid: Hyrax::CollectionType.find_or_create_admin_set_type.gid
+        }
+      end
+
+      def collection_metadata(collection)
+        {
+          title: [collection[:title]],
+          Bulkrax.system_identifier_field => collection[:id],
+          id: collection[:id],
+          visibility: 'open',
+          collection_type_gid: Hyrax::CollectionType.find_or_create_default_collection_type.gid
+        }
+      end
+
+      def collection_delimiter
+        Bulkrax.field_mappings["HykuAddons::CsvParser"]&.dig("collection", :split) || "\|"
+      end
+
+      def collection_prefix
```

Comment:
This not asking for anything but raising it for discussion. We are still on **bulkrax (0.1.0 82dffb2)** while the current Bulkax version is 2. Is the work a good opportunity to upgrade our bulkrax to version 2 and we could do remove this code.
The only other change after the upgrade will be for Tom M to update the csv files they use to use **parent** and **child** instead of the current collection mapping.

Reply:
I haven't added anything to test parent or child columns *in general* as that should be done when you update to Bulkrax.  All this does is test whether the particular parent column will work.  I think that updating to Bulkrax 3 should probably be raised as a separate ticket, so we can assess the priority of that.  This could be a good time to do it, I don't know.

Reply:
Thanks, I will ask other dev during Standup and then perhaps Elisa & Tom later since they might not know how far behind our version is, so they can decide since this feature is actively used.